lib/gemstash/gem_source/private_source.rb

```diff
@@ -67,7 +67,15 @@ def serve_info(name)
       end
 
       def serve_marshal(id)
-        halt 403, "Not yet supported"
+        gem_full_name = id.sub(/\.gemspec\.rz\z/, "")
+        gem = storage.resource(gem_full_name)
+        halt 404 unless gem.exist?
+        gem.load
+        halt 403, "That gem has been yanked" unless gem.properties[:indexed]
+        spec = spec_storage.resource(gem_full_name)
+        halt 404 unless spec.exist?
+        content_type "application/octet-stream"
+        spec.load.content
       end
```

Comment:
Does it make sense to extract this logic of finding the gemspec? looks like a lot of logic.

Also, what do you think of storing the content type into the gem spec properties so it is only required to forward the properties as we do with the gems?